packages/x-date-pickers/src/DateField/tests/selection.DateField.test.tsx

```diff
@@ -207,4 +207,24 @@ describe('<DateField /> - Selection', () => {
       expect(getCleanedSelectedContent(input)).to.equal('MM');
     });
   });
+
+  describe('key: Tab', () => {
+    it('should move selection to the next section when one section is selected', () => {
+      render(<DateField />);
+      const input = screen.getByRole('textbox');
+      clickOnInput(input, 'MM');
+      expect(getCleanedSelectedContent(input)).to.equal('MM');
+      userEvent.keyPress(input, { key: 'Tab' });
+      expect(getCleanedSelectedContent(input)).to.equal('DD');
+    });
+
+    it('should move selection to the previous section when one section is selected and Shift is pressed', () => {
```

Comment:
Maybe also add an "integration" test, checking that the focus can be moved between elements before the date field as well as after it.
You could render the following snippet to get natural tabulation order.
```tsx
<div>
  <button>before</button>
  <DateField />
  <button>after</button>
</div>
```